rdkit/Chem/Draw/UnitTestDraw.py

```diff
@@ -8,185 +7,154 @@
 
+  @unittest.skipIf(aggCanvas is None, 'Skipping agg test')
   def testAggFile(self):
-    try:
-      from rdkit.Chem.Draw.aggCanvas import Canvas
-    except ImportError:
-      logger.info("Skipping agg test")
-      return
     os.environ['RDKIT_CANVAS'] = 'agg'
+    self._testMolToFile()
 
-    foo, fn = tempfile.mkstemp(suffix='.png')
-    foo = None
-    self.assertEqual(os.path.getsize(fn), 0)
-
-    Draw.MolToFile(self.mol, fn)
-
-    self.assertNotEqual(os.path.getsize(fn), 0)
-    try:
-      os.unlink(fn)
-    except Exception:
-      pass
-
+  @unittest.skipIf(spingCanvas is None, 'Skipping sping test')
   def testSpingFile(self):
-    try:
-      from rdkit.Chem.Draw.spingCanvas import Canvas
-    except ImportError:
-      logger.info("Skipping sping test")
-      return
     os.environ['RDKIT_CANVAS'] = 'sping'
+    self._testMolToFile()
 
-    foo, fn = tempfile.mkstemp(suffix='.png')
-    foo = None
-    self.assertEqual(os.path.getsize(fn), 0)
-
-    Draw.MolToFile(self.mol, fn)
-
-    self.assertNotEqual(os.path.getsize(fn), 0)
-    try:
-      os.unlink(fn)
-    except Exception:
-      pass
-
-  def testCairoImage(self):
-    try:
-      from rdkit.Chem.Draw.cairoCanvas import Canvas
-    except ImportError:
-      return
-    os.environ['RDKIT_CANVAS'] = 'cairo'
-
-    img = Draw.MolToImage(self.mol, size=(300, 300))
+  def _testMolToImage(self, mol=None, kekulize=True, options=None, showImage=False, **kwargs):
+    mol = mol or self.mol
+    img = Draw.MolToImage(mol, size=(300, 300), kekulize=kekulize, options=options, **kwargs)
     self.assertTrue(img)
     self.assertEqual(img.size[0], 300)
     self.assertEqual(img.size[1], 300)
+    if self.showAllImages or showImage:
+      img.show()
+
+  @unittest.skipIf(cairoCanvas is None, 'Skipping cairo test')
+  def testCairoImage(self):
+    os.environ['RDKIT_CANVAS'] = 'cairo'
+    self._testMolToImage()
 
+  @unittest.skipIf(aggCanvas is None, 'Skipping agg test')
   def testAggImage(self):
-    try:
-      from rdkit.Chem.Draw.aggCanvas import Canvas
-    except ImportError:
-      return
     os.environ['RDKIT_CANVAS'] = 'agg'
-    img = Draw.MolToImage(self.mol, size=(300, 300))
-    self.assertTrue(img)
-    self.assertEqual(img.size[0], 300)
-    self.assertEqual(img.size[1], 300)
+    self._testMolToImage()
 
+  @unittest.skipIf(spingCanvas is None, 'Skipping sping test')
   def testSpingImage(self):
-    try:
-      from rdkit.Chem.Draw.spingCanvas import Canvas
-    except ImportError:
-      return
     os.environ['RDKIT_CANVAS'] = 'sping'
-    img = Draw.MolToImage(self.mol, size=(300, 300))
-    self.assertTrue(img)
-    self.assertEqual(img.size[0], 300)
-    self.assertEqual(img.size[1], 300)
+    self._testMolToImage()
 
+  @unittest.skipIf(qtCanvas is None, 'Skipping Qt test')
   def testQtImage(self):
-    import sys
     try:
       from PySide import QtGui
-      from rdkit.Chem.Draw.qtCanvas import Canvas
+      _ = QtGui.QApplication(sys.argv)
```

Comment:
Have you actually tried this on a machine with Qt installed? I don't know exactly how `_` works, but if the object is immediately there will be a problem. If there's no Qt application available when the rest of the code runs it will either crash or generate errors.